groupmne/inverse.py

```diff
@@ -7,24 +7,168 @@
+    n_subjects, n_channels, n_times = meeg.shape
+    n_features = gains_scaled.shape[-1]
+
+    # alpha is necessary for all models
+    if "alpha" not in estimator_kwargs.keys():
+        raise ValueError("The method %s requires the `alpha` hyperparameter."
+                         "See the documentation of %s for further detail."
+                         % (method, _method_to_str(method)))
+
+    # beta is necessary for dirty and ot models
+    if method not in ["lasso", "grouplasso", "relasso"]:
+        if "beta" not in estimator_kwargs.keys():
+            raise ValueError("The method %s requires the `beta`"
+                             " hyperparameter."
+                             "See the documentation of %s for further detail."
+                             % (method, _method_to_str(method)))
+
+    # ground metric and ot hyperparameters for ot models
+    if method in ["mtw", "remtw"]:
+        if "M" not in estimator_kwargs.keys():
+            raise ValueError("The method %s requires the OT ground metric `M`"
+                             " corresponding to the geodesic distance matrix "
+                             "over the cortical mantle. "
+                             "See the documentation of %s for further detail."
+                             % (method, _method_to_str(method)))
+        else:
+            M = estimator_kwargs["M"]
+            if len(M) != n_features or len(M.T) != n_features:
+                raise ValueError("The ground metric M must be an array"
+                                 "(n_features, n_features); got (%s, %s)"
+                                 % M.shape)
+            if M.min() < 0.:
+                raise ValueError("The ground metric M must be non-negative"
+                                 "got M.min() = %s"
+                                 % M.min())
+        if "gamma" not in estimator_kwargs.keys():
+            gamma = estimator_kwargs["M"].max()
+            estimator_kwargs["gamma"] = gamma
+        if "epsilon" not in estimator_kwargs.keys():
+            epsilon = 10. / n_features
+            estimator_kwargs["epsilon"] = epsilon
+
+    xty = np.array([g.T.dot(m) for g, m in zip(gains_scaled, meeg)])
+
+    # rescale l12 norm penalty
+    if method in ["grouplasso", "dirty"]:
+        # take L2 over subjects and max over time and space
+        # alphamax (n_times)
+        alphamax = np.linalg.norm(xty, axis=0).max() / n_channels
+        estimator_kwargs["alpha"] *= alphamax
+
+    # rescale l1 norm penalty
+    elif method in ["dirty", "mtw", "remtw", "lasso", "relasso"]:
+        betamax = abs(xty).max() / n_channels
+        if method in ["lasso", "relasso"]:
+            alpha_ = betamax * np.ones(n_subjects)
+            estimator_kwargs["alpha"] *= alpha_
+        else:
+            estimator_kwargs["beta"] *= betamax
+    return estimator_kwargs
+
+
+def compute_time_dependent_gl(gains, meeg, group_info, depth=0.9,
```

Comment:
is this meant to be a public function? Currently according to the doc the only public functions are:

   compute_fwd
   compute_inv_data
   compute_group_inverse

Reply:
Yes it should be made private